src/util/pair-checker.js

```diff
@@ -22,30 +22,67 @@ export function checkPair(context, { left, right }) {
      * @returns {{node, index}[]}
      */
     const foundMissingPairNodes = (currentStrInParagraph) => {
-        let foundLeft = false;
-        let matchParentheses = [];
-        currentStrInParagraph.forEach((node) => {
-            const text = getSource(node);
-            // left を探す
-            let leftIndex = -1;
-            if (!foundLeft) {
-                leftIndex = text.indexOf(left);
-                if (leftIndex !== -1) {
-                    matchParentheses.push({
+        let matchParentheses = currentStrInParagraph
+            .map((node) => {
+                let text = getSource(node);
+                let index = 0;
+                let symbolLocations = [];
+                while (index < text.length) {
+                    index = text.indexOf(left, index);
+                    if (index < 0) break;
+                    symbolLocations.push({
+                        index,
                         node,
-                        index: leftIndex
+                        type: "left"
                     });
-                    foundLeft = true;
+                    index += 1;
                 }
+                if (left !== right) {
+                    index = 0;
+                    while (index < text.length) {
+                        index = text.indexOf(right, index);
+                        if (index < 0) break;
+                        symbolLocations.push({
+                            index,
+                            node,
+                            type: "right"
+                        });
+                        index += 1;
+                    }
+                    symbolLocations.sort((a, b) => a.index - b.index);
+                }
+                return symbolLocations;
+            })
+            .flat();
+
+        if (left === right) {
+            if (matchParentheses.length % 2 == 0) {
+                return [];
+            } else {
+                return [matchParentheses[matchParentheses.length - 1]];
             }
-            // right を探す
-            let pairIndex = text.indexOf(right, leftIndex + 1);
-            if (pairIndex !== -1) {
-                matchParentheses.pop();
-                foundLeft = false;
+        } else {
+            let unmatchParences = [];
```

Comment:
```suggestion
            const unmatchParences = [];
```

popやshiftなのでconstでいいはず